android/jni/com/mapswithme/maps/editor/Editor.cpp

```diff
@@ -254,31 +260,57 @@ Java_com_mapswithme_maps_editor_Editor_nativeIsBuilding(JNIEnv * env, jclass cla
 JNIEXPORT jstring JNICALL
 Java_com_mapswithme_maps_editor_Editor_nativeGetDefaultName(JNIEnv * env, jclass)
 {
-  // TODO(yunikkk): add multilanguage names support via EditableMapObject::GetLocalizedName().
   return jni::ToJavaString(env, g_editableMapObject.GetDefaultName());
 }
 
 JNIEXPORT void JNICALL
 Java_com_mapswithme_maps_editor_Editor_nativeSetDefaultName(JNIEnv * env, jclass, jstring name)
 {
-  // TODO(yunikkk): add multilanguage names support.
   g_editableMapObject.SetName(jni::ToNativeString(env, name), StringUtf8Multilang::kDefaultCode);
 }
 
+JNIEXPORT jobjectArray JNICALL
+Java_com_mapswithme_maps_editor_Editor_nativeGetLocalizedNames(JNIEnv * env, jclass)
+{
+  return jni::ToJavaArray(env, g_localNameClazz, g_editableMapObject.GetLocalizedNames(), ToJavaName);
+}
+
+JNIEXPORT void JNICALL
+Java_com_mapswithme_maps_editor_Editor_nativeSetLocalizedNames(JNIEnv * env, jclass, jobjectArray names)
+{
+  int const length = env->GetArrayLength(names);
+  for (int i = 0; i < length; i++)
+  {
+    auto jName = env->GetObjectArrayElement(names, i);
```

Comment:
Можно, ок

> On Apr 28, 2016, at 12:06 PM, Alexander Marchuk <notifications@github.com> wrote:
> 
> In android/jni/com/mapswithme/maps/editor/Editor.cpp <https://github.com/mapsme/omim/pull/3042#discussion_r61395446>:
> 
> >    g_editableMapObject.SetName(jni::ToNativeString(env, name), StringUtf8Multilang::kDefaultCode);
> >  }
> >  
> > +JNIEXPORT jobjectArray JNICALL
> > +Java_com_mapswithme_maps_editor_Editor_nativeGetLocalizedNames(JNIEnv * env, jclass)
> > +{
> > +  return jni::ToJavaArray(env, g_localNameClazz, g_editableMapObject.GetLocalizedNames(), ToJavaName);
> > +}
> > +
> > +JNIEXPORT void JNICALL
> > +Java_com_mapswithme_maps_editor_Editor_nativeSetLocalizedNames(JNIEnv * env, jclass, jobjectArray names)
> > +{
> > +  int const length = env->GetArrayLength(names);
> > +  for (int i = 0; i < length; i++)
> > +  {
> > +    auto jName = env->GetObjectArrayElement(names, i);
> Вероятно, стоить сделать зеркальную функцию - из Java массива создавать вектор строкю
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub <https://github.com/mapsme/omim/pull/3042/files/bd670c25f99666f6d08e62a7563f971b86471553..63604f991b8a6a24f7d86c027282eb5cba964fac#r61395446>